### PR TITLE
replication subnet group

### PR DESCRIPTION
**Pull request by ruimoreira**
##### SUMMARY
New Module replication subnet group

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
replication subnet group
##### ADDITIONAL INFORMATION
replication subnet group - module to manage DMS subnet groups

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: dms_replication_subnet_group
+short_description: creates or destroys a data migration services subnet group
+description:
+    - creates or destroys a data migration services subnet group
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py]: This should be regular text.
```suggestion
    - Creates or destroys a data migration services subnet group.
```

File: lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Failed to Modify the DMS replication subnet group.",
+                         exception=traceback.format_exc(),
+                         **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Failed to Modify the DMS replication subnet group.",
+                         exception=traceback.format_exc())
+
+
+def delete_replication_subnet_group(connection):
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py]: What's this?

File: lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    return True
+
+
+def main():
+    argument_spec = dict(
+        state=dict(choices=['present', 'absent'], default='present'),
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py]: You should always have `type='str'` (or whatever type you have) in all of argument spec.

File: lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+        identifier=dict(required=True),
+        description=dict(required=True),
+        subnetids=dict(type='list', required=True),
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py]:
```suggestion
        subnetids=dict(type='list', required=True, elements='str'),
```
Except if you want the list elements to be of another type.

File: lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+    )
+    global module
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py]: I think it's not good style to use `global`.

File: lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+                    exit_message = dmsclient
+                changed = True
+            else:
+                exit_message = "No changes to Subnet group"
+        else:
+            if not module.check_mode:
+                exit_message = create_replication_subnet_group(dmsclient)
+                changed = True
+            else:
+                exit_message = "Check mode enabled"
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py:226]: `changed = True` should also happen in this branch.

File: lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+    elif state == 'absent':
+        if replication_subnet_exists(subnet_group):
+            replication_subnet_group_delete(dmsclient)
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py]: You should handle check mode here!

File: lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py (hunk 2 above)
Comment (ruimoreira) [reply lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py]: got missed, not really needed

**Pull request closed without merging**